geopandas/base.py

```diff
@@ -100,6 +100,14 @@ def _generate_sindex(self):
             _sindex = sindex_cls(self.geometry)
             if not _sindex.is_empty:
                 self._sindex = _sindex
+            else:
+                warn(
+                    "Future versions of GeoPandas will allow empty spatial indexes. "
+                    "Please use `len(gdf.sindex) > 0` or `if gdf.sindex` instead of "
+                    "`if gd.sindex is not None` to check for empty spatial indexes.",
```

Comment:
```suggestion
                    "Generated spatial index is empty and returned `None`. "
                    "Future versions of GeoPandas will return zero-length spatial "
                    "index instead of `None`. Use `len(gdf.sindex) > 0` "
                    "or `if gdf.sindex` instead of `if gd.sindex is not None` "
                    "to check for empty spatial indexes.",
```
This might be more explanatory. Or some variant of that.

Reply:
Sounds good to me, will change to that.